lib/Page.js

```diff
@@ -205,6 +288,7 @@ Page.prototype.generate = function (builtFiles) {
         this.collectFrontMatter(result);
         return this.removeFrontMatter(result);
       })
+      .then(result => this.insertSiteNav((result))) // Ensure this is before insertFooter
```

Comment:
The comment can be inferred from the code. :P

Reply:
This comment was requested from @acjh for easy reference.

Reply:
Perhaps explain it concisely (on its own line if need be) as the rationale is non-obvious?